map.go

```diff
@@ -1,21 +1,27 @@
 package indexmap
 
-import "sync"
+import (
+	"sort"
+	"sync"
+
+	"golang.org/x/exp/constraints"
+)
 
 // IndexMap is a map supports seeking data with more indexes.
 // Serializing a IndexMap as JSON results in the same as serializing a map,
 // the result doesn't contain the index information, only data.
 // NOTE: DO NOT insert nil value into the IndexMap
-type IndexMap[K comparable, V any] struct {
+type IndexMap[K constraints.Ordered, V any] struct {
```

Comment:
IMO, this is too narrow of a constraint (no pun intended) on possible key types. When using `Ordered`, arrays and comparable structs would not be accepted as keys. IndexMap is supposed to be like a map, but with additional features, so it should accept all key types that a normal map can have. Maybe you should instead make a wrapper around IndexMap for whatever you're trying to achieve.